compiler/rustc_infer/src/infer/mod.rs

```diff
@@ -945,14 +945,20 @@ impl<'tcx> InferCtxt<'tcx> {
             (&ty::Infer(ty::TyVar(a_vid)), &ty::Infer(ty::TyVar(b_vid))) => {
                 return Err((a_vid, b_vid));
             }
+            (ty::Alias(ty::Opaque, ..), ty::Infer(ty::TyVar(_)))
+            | (ty::Infer(ty::TyVar(_)), ty::Alias(ty::Opaque, ..)) => {}
+            (ty::Alias(ty::Opaque, ..), _) | (_, ty::Alias(ty::Opaque, ..)) => span_bug!(
+                cause.span(),
+                "opaque types got hidden types registered from within subtype predicate: {r_a:?} vs {r_b:?}"
+            ),
```

Comment:
This doesn't cover all cases imo. It really should be something like: if lhs has infers, then rhs is a ty var, or vice versa.

Reply:
could still constrain, e.g., if `W<impl Sized>` is equated to `W<i32>.`